app/Http/Controllers/BookController.php

```diff
@@ -89,7 +105,51 @@ public function claimBookById(Int $id, Request $request)
             'message' => "Book $id was claimed"
         ], 200);
         }
+    }
+
+    public function returnBookById(Int $id, Request $request)
+    {
+        try {
+            $book = $this->book->findOrFail($id);
+        } catch (Exception $e) {
+            return response()->json([
+                'message' => "Book $id was not found"
+            ], 404);
+        }
+
+        // Validating the request data
+        $request->validate([
+            'email' => 'required|email',
+        ]);
+
+        $book->claim;
+
+        if(is_null($book->claim)) {
+            return response()->json([
+                'message' => "Book $id is not currently claimed"
+            ], 400);
+        }
+
+        $claim = $this->claim->where('book_id', '=', $id)->firstOrFail();
```

Comment:
You should probs put this in a try catch, if for some reason the claim doesn't exist, at the moment an exception will crash the app.

Put it in a try catch, and return the exact same response as your 500 response

Reply:
We have now put this into a try catch as recommended.